service/metrics.js

```diff
@@ -1,109 +1,36 @@
+			if (err) {
+				console.error(err);
+			}
+		});
+	}, reportInterval);
 }
 
-/** Increment the counter for response type where type is '.js' or '.css' */
-ServiceMetrics.prototype.addResponseType = function(type) {
-	if (type === ".js") {
-		this.javascriptResponsesCount++;
-		this.javascriptResponsesLastUpdated = Date.now();
-		return;
-	}
-
-	if (type === ".css") {
-		this.cssResponsesCount++;
-		this.cssResponsesLastUpdated = Date.now();
-		return;
-	}
-
-	throw new Error("Invalid response file type: " + type);
-};
-
-ServiceMetrics.prototype.addResponseTime = function(timeInMilliSecs) {
-	var newaverage = (timeInMilliSecs + (this.averageResponseTime * this.responseCount)) / (this.responseCount + 1);
-
-	if (timeInMilliSecs < this.minResponseTime) {
-		this.minResponseTime = timeInMilliSecs;
-	}
-
-	if (timeInMilliSecs > this.maxResponseTime) {
-		this.maxResponseTime = timeInMilliSecs;
-	}
-
-	this.averageResponseTime = newaverage;
-	this.responseCount++;
-	this.lastResponseAverageUpdate = Date.now();
-};
-
-ServiceMetrics.prototype.getResponseTimeMetric = function() {
-	return {
-		"type": "movingaverage",
-		"description": "Average application response time",
-		"since": new Date(this.serviceStartTime).toISOString(),
-		"period": Math.floor((Date.now() - this.serviceStartTime) / 1000),
-		"mean": this.averageResponseTime,
-		"min": this.minResponseTime,
-		"max": this.maxResponseTime,
-		"lastUpdate": this.lastResponseAverageUpdate == null ? null : new Date(this.lastResponseAverageUpdate).toISOString(),
-		"unit": "milliseconds"
-	};
-};
-
-ServiceMetrics.prototype.getUptimeMetric = function() {
-	return {
-		"type": "counter",
-		"description": "The time since the Node JS process was last restarted",
-		"val": Math.floor((Date.now() - this.serviceStartTime) / 1000),
-		"unit": "seconds",
-		"since": new Date(this.serviceStartTime).toISOString(),
-		"lastUpdated": new Date().toISOString()
-	};
-};
-
-ServiceMetrics.prototype.getJavascriptResponseCountMetric = function() {
-	return {
-		"type": "counter",
-		"description": "The number of Javascript polyfills served since last Node JS Process restart",
-		"val": this.javascriptResponsesCount,
-		"unit": "items",
-		"since": new Date(this.serviceStartTime).toISOString(),
-		"lastUpdated": new Date(this.javascriptResponsesLastUpdated).toISOString()
-	};
-};
-
-ServiceMetrics.prototype.getCSSResponseCountMetric = function() {
-	return {
-		"type": "counter",
-		"description": "The number of CSS polyfills served since last Node JS Process restart",
-		"val": this.cssResponsesCount,
-		"unit": "items",
-		"since": new Date(this.serviceStartTime).toISOString(),
-		"lastUpdated": new Date(this.cssResponsesLastUpdated).toISOString()
-	};
-};
+// This could be added to node-measured, perhaps as a static 
+// Measured.flattenMetrics(instance.getJSON()), or by adding a 
+// boolean arg to getJSON(). Consider making a PR.
+function flatten(obj, prefix, root) {
```

Comment:
This looks like the [same algorithm](https://github.com/felixge/node-graphite/blob/master/lib/GraphiteClient.js#L15) which is called by [`graphite.write`](https://github.com/felixge/node-graphite/blob/master/lib/GraphiteClient.js#L15), I don't think this is required.